haystack/dataclasses/document.py

```diff
@@ -47,6 +44,10 @@ def __call__(cls, *args, **kwargs):
         if "id_hash_keys" in kwargs:
             del kwargs["id_hash_keys"]
 
+        ### >=2.11: Dataframe is not supported anymore ###
+        if "dataframe" in kwargs:
+            raise ValueError("The `dataframe` field is no longer supported.")
```

Comment:
If we want to raise this ValueError, I agree with you that we should also have a test case that covers it. I am just wondering if we really need this ValueError raised here. It contains no link to more info and no instructions what to do to migrate.
Without this ValueError, `TypeError: Document.__init__() got an unexpected keyword argument 'dataframe'` is what users will get. Should be enough?

Reply:
Makes sense. Done in https://github.com/deepset-ai/haystack/pull/8906/commits/bf5b84cbe3a1102bfd15c5dfa1aeea61f7660741